packages/harpguru-core/src/hooks/use-viewable-matrices/use-viewable-matrices.test.ts

```diff
@@ -0,0 +1,197 @@
+    [Root, Second, Third],
+    [Second, Third, Root],
+    [Third, Second, Root],
+  ]
+  const modifiedHarpStrata = {
+    ...inactiveCellsHarpStrata,
+    degreeMatrix: simplifiedDegreeMatrix,
+  }
+  const expectedViewableMatrix = [[Root], [Second], [Third]]
+  mockUseGlobal.mockImplementation((stateItem: string) => {
+    if (stateItem === 'activeHarpStrata') return [modifiedHarpStrata]
+    if (stateItem === 'columnBounds') return [[0, 0]]
+    return undefined
+  })
+
+  expect(useViewableMatrices().viewableDegreeMatrix).toStrictEqual(
+    expectedViewableMatrix
+  )
+})
+
+test('Middle columns of degreeMatrix are viewable when columnBounds is [1, 2]', () => {
+  const { Root, Second, Third } = DegreeIds
+  const simplifiedDegreeMatrix = [
+    [Second, Root, Second, Third],
+    [Second, Second, Third, Root],
+    [Second, Third, Second, Root],
+  ]
+  const modifiedHarpStrata = {
+    ...inactiveCellsHarpStrata,
+    degreeMatrix: simplifiedDegreeMatrix,
+  }
+  const expectedViewableMatrix = [
+    [Root, Second],
+    [Second, Third],
+    [Third, Second],
+  ]
+  mockUseGlobal.mockImplementation((stateItem: string) => {
+    if (stateItem === 'activeHarpStrata') return [modifiedHarpStrata]
+    if (stateItem === 'columnBounds') return [[1, 2]]
+    return undefined
+  })
+
+  expect(useViewableMatrices().viewableDegreeMatrix).toStrictEqual(
+    expectedViewableMatrix
+  )
+})
+
+test('Blank rows are removed from the viewable matrix even if the broader context isnt empty', () => {
+  const { Root, Second, Third } = DegreeIds
+  const simplifiedDegreeMatrix = [
+    [Second, undefined, undefined, undefined],
+    [Second, Root, Second, Third],
+    [Second, Second, Third, Root],
+    [Second, Third, Second, Root],
+    [Second, Third, Second, Root],
+    [undefined, undefined, undefined, Second],
+  ]
+  const modifiedHarpStrata = {
+    ...inactiveCellsHarpStrata,
+    degreeMatrix: simplifiedDegreeMatrix,
+  }
+  const expectedViewableMatrix = [
+    [Root, Second],
+    [Second, Third],
+    [Third, Second],
+    [Third, Second],
+  ]
+  mockUseGlobal.mockImplementation((stateItem: string) => {
+    if (stateItem === 'activeHarpStrata') return [modifiedHarpStrata]
+    if (stateItem === 'columnBounds') return [[1, 2]]
+    return undefined
+  })
+
+  expect(useViewableMatrices().viewableDegreeMatrix).toStrictEqual(
+    expectedViewableMatrix
+  )
+})
+
+test('Entire pitchMatrix is viewable when columnBounds is FIT', () => {
+  mockUseGlobal.mockImplementation((stateItem: string) => {
+    if (stateItem === 'activeHarpStrata') return [inactiveCellsHarpStrata]
+    if (stateItem === 'columnBounds') return ['FIT']
+    return undefined
+  })
+
+  expect(useViewableMatrices().viewablePitchMatrix).toStrictEqual(
+    inactiveCellsHarpStrata.pitchMatrix
+  )
+})
+
+test('Bounded pitch matrix can be returned', () => {
+  const { A, B, C } = PitchIds
+  // TODO: This is actually subverting the type checking.
```

Comment:
TODO here

Reply:
Raised as techdebt in https://github.com/js-jslog/harpguru/issues/113#issuecomment-922351622

The work done to make the app sizing dynamic (at #107) was used as a platform for additional speculative pull requests to ensure that this was the correct direction to start down.

Rather than addressing the finer details in this or subsequent pull request and potentially struggling to merge up to those subsequent branches I will prefer to record the issues in each merge request and resolve them afterwards.

This has a limit of course. Major issues will not be merged to master. I will need to find a way around that if it happens.